watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -1199,8 +1240,135 @@ def rule_visc_k_phase(b, p):
     def _radius_stokes_comp(self):
         add_object_reference(self, "radius_stokes_comp", self.params.radius_stokes_comp)
 
+    def _molar_volume_phase_comp(self):
+        add_object_reference(
+            self, "molar_volume_phase_comp", self.params.molar_volume_phase_comp
+        )
+
     def _diffus_phase_comp(self):
-        add_object_reference(self, "diffus_phase_comp", self.params.diffus_phase_comp)
+        # Retrieve two components sets from the molar_volume_data and diffusivity_data configurations
+        self.mv_dt_ind = Set(
+            initialize=[i[1] for i in self.params.config.molar_volume_data.keys()]
+        )
+        self.diffus_dt_ind = Set(
+            initialize=[i[1] for i in self.params.config.diffusivity_data.keys()]
+        )
+        if self.params.config.diffus_calculation == DiffusivityCalculation.none:
+            missing_diffus_ind = [
+                i
+                for i in (self.params.ion_set | self.params.solute_set)
+                if i not in self.diffus_dt_ind
+            ]
+            if not missing_diffus_ind == []:
+                _log.warning(
+                    "Diffusivity data is not provided for {}.".format(
+                        missing_diffus_ind
+                    )
+                )  # warning for components with no diffusivity_data entry.
+            self.diffus_phase_comp = Var(
+                self.params.phase_list,
+                self.diffus_dt_ind,
+                initialize=1e-9,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="mass diffusivity of solute components",
+            )
+        elif (
+            self.params.config.diffus_calculation == DiffusivityCalculation.HaydukLaudie
+        ):
+            missing_diffus_ind = [
+                i
+                for i in (self.params.ion_set | self.params.solute_set)
+                if i not in (self.mv_dt_ind | self.diffus_dt_ind)
+            ]
+            if not missing_diffus_ind == []:
+                _log.warning(
+                    "No diffusivity_data or molar_volume_data is provided for {}; "
+                    "there will be no diffus_phase_comp properties for these components.".format(
+                        missing_diffus_ind
+                    )
+                )  # warning for components with neither diffusivity_data nor molar_volume_data entry.
+            common_ind = [i for i in self.mv_dt_ind if i in self.diffus_dt_ind]
+            _log.warning(
+                "Both diffusivity_data and molar_volume_data are provided for {}; "
+                "the diffus_phase_comp property will be calculated based on their molar_volume_data "
+                "for these components becuase the HaydukLadie method is selected.".format(
```

Comment:
```suggestion
                "for these components because the Hayduk-Ladie method is selected.".format(
```

Reply:
The `-` was not written to be consistent to the name in the Method Class, where it is named HaydukLadie, I would keep them consistent regardless of having - or not

Reply:
```suggestion
                "for these components becuase the HaydukLaudie method is selected.".format(
```
Fair enough- still needs a "u" though.